azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/spring/SpringDeploymentClient.java

```diff
@@ -121,7 +128,8 @@ private DeploymentResourceInner createDeployment(Deployment deploymentConfigurat
     private DeploymentResourceInner updateDeployment(Deployment deploymentConfiguration, DeploymentResourceInner deployment,
                                                      ResourceUploadDefinitionInner resource) throws AzureExecutionException {
         final DeploymentSettings previousDeploymentSettings = deployment.properties().deploymentSettings();
-        if (isResourceScaled(deploymentConfiguration, previousDeploymentSettings)) {
+        final SkuInner sku = deployment.sku();
+        if (isResourceScaled(deploymentConfiguration, previousDeploymentSettings, sku)) {
```

Comment:
Could we pass the `DeploymentResourceInner` directly instead of pass `DeploymentSettings` and `Sku`